metaflow/plugins/airflow/airflow_utils.py

```diff
@@ -0,0 +1,410 @@
+
+
+def run_id_creator(val):
+    # join `[dag-id,run-id]` of airflow dag.
+    return hashlib.md5("-".join(val).encode("utf-8")).hexdigest()[:RUN_ID_LEN]
+
+
+def task_id_creator(lst):
+    # This is a filter which creates a hash of the run_id/step_name string.
+    # Since run_ids in airflow are constants, they don't create an issue with the
+    #
+    return hashlib.md5("/".join(lst).encode("utf-8")).hexdigest()[:TASK_ID_LEN]
+
+
+def json_dump(val):
+    return json.dumps(val)
+
+
+class AirflowDAGArgs(object):
+
+    # `_arg_types` is a dictionary which represents the types of the arguments of an Airflow `DAG`.
+    # `_arg_types` is used when parsing types back from the configuration json.
+    # It doesn't cover all the arguments but covers many of the important one which can come from the cli.
+    _arg_types = {
+        "dag_id": str,
+        "description": str,
+        "schedule_interval": str,
+        "start_date": datetime,
+        "catchup": bool,
+        "tags": list,
+        "dagrun_timeout": timedelta,
+        "default_args": {
+            "owner": str,
+            "depends_on_past": bool,
+            "email": list,
+            "email_on_failure": bool,
+            "email_on_retry": bool,
+            "retries": int,
+            "retry_delay": timedelta,
+            "queue": str,  #  which queue to target when running this job. Not all executors implement queue management, the CeleryExecutor does support targeting specific queues.
+            "pool": str,  # the slot pool this task should run in, slot pools are a way to limit concurrency for certain tasks
+            "priority_weight": int,
+            "wait_for_downstream": bool,
+            "sla": timedelta,
+            "execution_timeout": timedelta,
+            "trigger_rule": str,
+        },
+    }
+
+    # Reference for user_defined_filters : https://stackoverflow.com/a/70175317
+    filters = dict(
+        task_id_creator=lambda v: task_id_creator(v),
+        json_dump=lambda val: json_dump(val),
+        run_id_creator=lambda val: run_id_creator(val),
+    )
+
+    def __init__(self, **kwargs):
+        self._args = kwargs
+
+    @property
+    def arguments(self):
+        return dict(**self._args, user_defined_filters=self.filters)
+
+    def serialize(self):
+        def parse_args(dd):
+            data_dict = {}
+            for k, v in dd.items():
+                # see the comment below for `from_dict`
+                if isinstance(v, dict):
+                    data_dict[k] = parse_args(v)
+                elif isinstance(v, datetime):
+                    data_dict[k] = v.isoformat()
+                elif isinstance(v, timedelta):
+                    data_dict[k] = dict(seconds=v.total_seconds())
+                else:
+                    data_dict[k] = v
+            return data_dict
+
+        return parse_args(self._args)
+
+    @classmethod
+    def deserialize(cls, data_dict):
+        def parse_args(dd, type_check_dict):
```

Comment:
this can be simplified further? It seems that we have just one element which is `timedelta`.

Reply:
we have more than one `timedelta` / `datetime` elements. Simplified these functions a little.

Reply:
Check if `start_date` is needed.